copilot-instructions.md

```diff
@@ -0,0 +1,232 @@
+# Copilot Instructions for .NET Android Libraries Repository
+
+## Repository Overview
+
+This repository creates and maintains .NET for Android bindings for Google's Java/Kotlin Android libraries, including:
+- AndroidX libraries (600+ packages)
+- Google Play Services 
+- Firebase
+- ML Kit
+- Various third-party dependencies
+
+The repository uses a **config-driven approach** where all bindings are defined in `config.json` and automatically generated using the "binderator" tool.
+
+## Key Architecture Components
+
+### Core Files
+- **`config.json`**: Master configuration containing all 600+ Maven artifacts to bind, their versions, and NuGet package information
+- **`build.cake`**: Main Cake build script orchestrating the entire build process
+- **`BUILDING.md`**: Comprehensive build instructions and prerequisites
+- **`global.json`**: .NET SDK version requirements (currently .NET 8.0.408)
+
+### Directory Structure
+- **`source/`**: Contains binding customizations organized by Maven groupId (e.g., `androidx.core`, `com.google.android.gms`) and Razor templates (`.cshtml` files)
+- **`util/Xamarin.AndroidBinderator/`**: The "binderator" tool that generates binding projects from config
+- **`generated/`**: Auto-generated binding projects (created during build, not in source control)
+- **`docs/`**: Additional documentation including artifact lists and development tips
+
+## Build System
+
+### Prerequisites
+- .NET SDK 8.0.404+ (specified in global.json)
+- Cake .NET Tool: `dotnet tool install -g cake.tool`
+- Microsoft OpenJDK 11 (JDK-17 not supported)
+- Android SDK with API-29+ and `ANDROID_SDK_ROOT` environment variable
+- Optional: `api-tools` for API diffs: `dotnet tool install -g api-tools`
+
+### Common Build Commands
+```bash
+# Full build of all packages
+dotnet cake
+
+# Clean build for CI
+dotnet cake -t=ci
+
+# Build specific target
+dotnet cake --target=libs
+dotnet cake --target=nuget
+dotnet cake --target=packages
+
+# Update config to latest Maven versions
+dotnet cake --target=update-config
+
+# Bump NuGet package versions (4th component)
+dotnet cake --target=bump-config
+
+# Generate API differences
+dotnet cake nuget-diff.cake
+
+# Run utilities (governance, mappings, etc.)
+dotnet cake utilities.cake
+```
+
+## Configuration System
+
+### config.json Structure
```

Comment:
Make a note here that we normally wouldn't edit this file manually and use cake steps to do it.